rust/src/buffer.rs

```diff
@@ -32,6 +32,7 @@ pub struct Buffer<T> {
 }
 
 impl<T> Buffer<T> {
+    /// create a buffer from an existing region of memory (must already be byte-aligned)
     pub fn from_raw_parts(data: *const T, len: i32) -> Self {
```

Comment:
I believe this should be marked unsafe - if an incorrect pointer or len is passed, `get` and `slice` could access arbitrary memory.